Add an internal setting to track whether closure might be run

kripken opened the pull request:
This new internal setting is true if we run it, or if the user might run it.

Use that new setting to gate when to add closure annotations, which should
fix remaining issues with closure in emcc working but not after it.

cc @walkingeyerobot

sbc100 commented on src/settings_internal.js:
```diff
@@ -238,6 +238,11 @@ var HAS_MAIN = false;
+// Set when closure compiler may be run: Either emcc will run it, or the user
+// might run it after emcc. Either way, some JS changes and annotations must be
+// emitted in that case for closure compiler.
+var MAYBE_CLOSURE_COMPILER = false;
```
How about calling this something like "CLOSURE_COMPATIBLE"?    I'm not sure why we need MAYBE in the name.

I guess `--minify=0` isn't quite the same as `--closure-compatible` but we could always add the latter later if we want.. but the internal name I think doesn't need maybe in it does it?

kripken replied:
We have other `MAYBE_*` uses in the codebase, like `MAYBE_WASM2JS`, so this is consistent with that.

I think it's unambiguous what "maybe" means? But changing them all to "compatible" makes equal sense to me I guess. That is, as the comment says this setting is true when closure might be run (which is separate from using `minify=0` to determine that - we currently use that as an indicator, but we might change that later. if we do, this setting and uses of it will still make sense with no rewriting except for the code that sets the setting or not)

sbc100 replied:
Since its internal I don't really mind.   lgtm either way

kripken commented:
Test failures show I was not careful enough. The `shell` files change broke some things, because it didn't just add a comment but also changed how the `Module` object is received - and we can only assume closure will definitely run if `USE_CLOSURE` and not `MAYBE_CLOSURE`.

Those places might need some form of fix for the maybe case, I'm not sure, but I'll look into that separately, and removed it from this PR.
